rest_docs/rest-catalog-open-api.yaml

```diff
@@ -0,0 +1,849 @@
+
+  /v1/namespaces/{namespace}/tables:
+    parameters:
+      - $ref: '#/components/parameters/namespace'
+    get:
+      tags:
+        - Catalog API
+      summary: List all table identifiers underneath a given namespace
+      description: Return all table identifiers under this namespace
+      operationId: listTables
+      responses:
+        200:
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/responses/ListTablesResponse'
+              examples:
+                NonEmptyResponse:
+                  $ref: '#/components/examples/ListTablesNonEmptyExample'
+                EmptyNamespaceResponse:
+                  $ref: '#/components/examples/ListTablesEmptyExample'
+        401:
+          $ref: '#/components/responses/UnauthorizedResponse'
+        404:
+          description: Not Found - The namespace specified does not exist
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/responses/IcebergErrorResponse'
+              examples:
+                NamespaceNotFound:
+                  $ref: '#/components/examples/NoSuchNamespaceError'
+
+  /v1/namespaces/{namespace}/tables/{table}:
+    parameters:
+      - $ref: '#/components/parameters/namespace'
+      - $ref: '#/components/parameters/table'
+    delete:
+      tags:
+        - Catalog API
+      summary: Drop a table from the catalog
+      operationId: dropTable
+      description: Remove a table from the catalog
+      parameters:
+        - name: purgeRequested
+          in: query
+          required: false
+          description: Whether the user requested to purge the underlying table's data and metadata
+          schema:
+            type: boolean
+            default: false
+      responses:
+        200:
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/responses/DropTableResponse'
+              example: { "dropped": true, "purged": false }
+        202:
+          description: Accepted - for use if purgeRequested is implemented as an asynchronous action.
+        401:
+          $ref: '#/components/responses/UnauthorizedResponse'
+        404:
+          description:
+            Not Found - NoSuchTableException, Table to drop does not exist
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/responses/IcebergErrorResponse'
+              examples:
+                TableToRenameDoesNotExist:
+                  $ref: '#/components/examples/NoSuchTableError'
+    head:
+      tags:
+        - Catalog API
+      summary: Check if a table exists
+      operationId: tableExists
+      description:
+        Check if a table exists within a given namespace. This request does not return a response body.
+      responses:
+        200:
+          description: OK - Table Exists
```

Comment:
@rymurr and @jackye1995, what do you think about adding a body to the HEAD responses? I think it would be nice to do for consistency.

As an alternative, maybe we should avoid HEAD routes so that we don't have to deviate? We might not actually need this since the GET route isn't expensive. It just returns the namespace metadata.

Reply:
in that case, I feel it's better to use POST `/v1/tables/head` to express such operation, otherwise I am not sure what is going to happen for an open-api generated client.

Reply:
Wouldn't it be better to use a `GET` request to `/v1/namespaces/{namespace}/tables/{table}/exists` instead?

But I don't think that's worth it. Let's just leave this as-is.

Reply:
I think it's fine as-is for now. But a lot of libraries seem to allow bodies for HEAD requests. We should consider it in the future though. I know there are "the standard(s)" but quite a number of well used libraries do seem to deviate from them now and then for practical purposes.